handlers/handlers.go

```diff
@@ -1,238 +1,238 @@
-			if err != nil {
-				panic(err)
-			}
-
-			expirationTime := time.Now().Add(credentialTimeoutInMinutes * time.Minute)
-
-			cookie := http.Cookie{
-				Name:     cerealNotesCookieName,
-				Value:    token,
-				Expires:  expirationTime,
-				HttpOnly: true,
-			}
-
-			http.SetCookie(responseWriter, &cookie)
-		}
-
-		responseWriter.WriteHeader(http.StatusCreated)
-		responseWriter.Write([]byte(fmt.Sprint("passward email combo was correct")))
-
-	case http.MethodDelete:
-		// Cookie will overwrite existing cookie then delete itself
-		cookie := http.Cookie{
-			Name:     cerealNotesCookieName,
-			Value:    "",
-			HttpOnly: true,
-			MaxAge:   -1,
-		}
-
-		http.SetCookie(responseWriter, &cookie)
-		responseWriter.WriteHeader(http.StatusOK)
-		responseWriter.Write([]byte(fmt.Sprint("user succefully logged out")))
-	default:
-		respondWithMethodNotAllowed(
-			responseWriter,
-			[]string{http.MethodPost, http.MethodDelete})
-	}
-}
-
-type AuthentictedRequestHandlerType func(
-	http.ResponseWriter,
-	*http.Request,
-	models.UserId)
-
-func AuthenticateOrRedirectToLogin(
-	authenticatedHandlerFunc AuthentictedRequestHandlerType,
-) func(http.ResponseWriter, *http.Request) {
-	return func(responseWriter http.ResponseWriter, request *http.Request) {
-			if userId, err := getUserIdFromJwtToken(request); err != nil {
-				// If not loggedin redirect to login page
-				http.Redirect(
-					responseWriter,
-					request,
-					paths.LoginOrSignupPath,
-					http.StatusTemporaryRedirect)
-			} else {
-				authenticatedHandlerFunc(responseWriter, request, userId)
-			}
-		}
-}
-
-// Authenticated Handlers
-func HandleHomeRequest(
-	responseWriter http.ResponseWriter,
-	request *http.Request,
-	userId models.UserId,
-) {
-	switch request.Method {
-	case http.MethodGet:
-		parsedTemplate, err := template.ParseFiles("templates/home.tmpl")
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		parsedTemplate.Execute(responseWriter, userId)
-	default:
-		respondWithMethodNotAllowed(responseWriter, []string{http.MethodGet})
-	}
-}
-
-// UTIL
-func respondWithMethodNotAllowed(
-	responseWriter http.ResponseWriter,
-	allowedMethods []string,
-) {
-	allowedMethodsString := strings.Join(allowedMethods, ", ")
-	responseWriter.Header().Set("Allow", allowedMethodsString)
-
-	statusCode := http.StatusMethodNotAllowed
-	http.Error(responseWriter, http.StatusText(statusCode), statusCode)
-}
-
-func getRequestBody(request *http.Request) []byte {
-	body, err := ioutil.ReadAll(request.Body)
-	if err != nil {
-		panic(err)
-	}
-
-	if err := request.Body.Close(); err != nil {
-		panic(err)
-	}
-
-	return body
-}
+package handlers 
+ 
+import ( 
+	"encoding/json" 
+	"fmt" 
+	"github.com/atmiguel/cerealnotes/models" 
+	"github.com/atmiguel/cerealnotes/paths" 
+	"github.com/atmiguel/cerealnotes/services/userservice" 
+	"github.com/dgrijalva/jwt-go" 
+	"html/template" 
+	"io/ioutil" 
+	"log" 
+	"net/http" 
+	"strings" 
+	"time" 
+) 
+ 
+const oneWeekInMinutes = 60 * 24 * 7 
```

Comment:
Probably makes more sense for this to be a [time.Duration](https://golang.org/pkg/time/#Duration) instead of an int.

`const oneWeek = time.Hour * 24 * 7`

You can get the number of minutes that this duration is by doing `duration.Minutes()`.
Using `time.Duration` gives you the standard advantages that a time library gives you. Lets you do time arithmetic, lets you see if one time comes before or after another, better formatting when printing times ... stuff like that

Reply:
When you end up using this in`tokenutil`, you do something like
```go
jwt.StandardClaims{
   ExpiresAt: time.Now().Add(oneWeek).Unix()
}
```